sdk/txbuild/formation.go

```diff
@@ -35,6 +37,13 @@ func Formation(p FormationParams) (*txnbuild.Transaction, error) {
 		SourceAccount: p.InitiatorEscrow.Address(),
 		Signer:        &txnbuild.Signer{Address: p.InitiatorSigner.Address(), Weight: 1},
 	})
+	if !p.Asset.IsNative() {
+		tp.Operations = append(tp.Operations, &txnbuild.ChangeTrust{
+			Line:          p.Asset,
+			Limit:         p.AssetLimit,
+			SourceAccount: p.InitiatorEscrow.Address(),
+		})
+	}
 	tp.Operations = append(tp.Operations, &txnbuild.EndSponsoringFutureReserves{SourceAccount: p.InitiatorEscrow.Address()})
 	tp.Operations = append(tp.Operations, &txnbuild.BeginSponsoringFutureReserves{SourceAccount: p.ResponderSigner.Address(), SponsoredID: p.ResponderEscrow.Address()})
```

Comment:
my view is avoiding over-modularizing code at this early stage in the project. Right now it may make sense to group this code into a helper method, but things are changing pretty frequently. Imo a helper method here (called just twice, doesn't really make it that much more readable) right now may end up being more work in the long run trying to fit helper methods to use cases when things change, or then removing the helper methods since they don't make sense anymore

if alright with you I'd like to leave as is. As the project becomes more set in stone is when I think it would make sense to make these changes.

lmk if you disagree though happy to discuss / adjust my coding methodology 😃

Reply:
Yeah, that makes sense. I'm generally thinking in terms of descriptive logical units but it's equally good to leave as is so I'm fine with that. Thanks for the context!